tools/clang/lib/SPIRV/EmitSPIRVAction.cpp

```diff
@@ -365,12 +361,33 @@ class SPIRVEmitter : public ASTConsumer {
       return translateAPInt(intLiteral->getValue(), expr->getType());
     } else if (auto *floatLiteral = dyn_cast<FloatingLiteral>(expr)) {
       return translateAPFloat(floatLiteral->getValue(), expr->getType());
+    } else if (auto *binOp = dyn_cast<BinaryOperator>(expr)) {
+      return doBinaryOperator(binOp);
     }
+
     emitError("Expr '%0' is not supported yet.") << expr->getStmtClassName();
     // TODO: handle other expressions
     return 0;
   }
 
+  uint32_t doBinaryOperator(const BinaryOperator *expr) {
+    const auto opcode = expr->getOpcode();
+    const uint32_t lhs = doExpr(expr->getLHS());
```

Comment:
I would find it more intuitive if the code for lhs were generated *after* the code for the *rhs*.

Both Glslang and Clang will do it that way.

This starts to matter when you start accessing the insides of composite objects.